### PR TITLE
Feature/#33

### PR DESCRIPTION
**Pull request by dorusomcutean**

### CODE REVIEW

File: routes/orders/order.model.js [hunk 1]
```diff
@@ -103,9 +108,26 @@ OrderSchema.statics = {
...
+        return this.find({ status: orderStatus })
+            .sort({ createdAt: -1 })
             .exec();
+    },
+
+    addItems(id, items) {
+        return this.findOneAndUpdate(
+            { "_id": id },
+            { $push: { items: { $each: items } } },
+            { new: true }
```
Comment (last1here) [inline routes/orders/order.model.js:120]: Not sure what this `new` is doesnt seem to be [documented](https://docs.mongodb.com/manual/reference/method/db.collection.findOneAndUpdate/#definition).

File: client/app/shared/order.service.ts [hunk 2]
```diff
@@ -23,13 +23,36 @@ export class OrderService {
...
+        return this.http.post('/api/orders/' + orderId + '/complete', discountedValue)
+            .pipe(
+                catchError((error: any) => observableThrowError(error))
+            );
+    }
+
+    /**
+     * Calls API endpoint to update an order
+     */
+    updateOrder(orderId, items): Observable<any> {
```
Comment (last1here) [inline client/app/shared/order.service.ts]: Type function params.

File: client/app/order-panel/order-panel.component.ts [hunk 3]
```diff
@@ -82,17 +89,81 @@ export class OrderPanelComponent implements OnInit {
         if (this.discountForm === undefined) {
             return;
         }
-        const discountValue = this.discountForm.value.discount;
-        this.dicountedValue = (discountValue / 100) * this.totalNoDiscount;
 
-        this.totalWithDiscount = this.totalNoDiscount - this.dicountedValue;
+        if (this.order.discount !== undefined && this.discountForm.value.discount === 0) {
```
Comment (last1here) [inline client/app/order-panel/order-panel.component.ts:89]: Is there a reason to store the discounted value outside of `this.order.discount`?

File: routes/orders/order.model.js (hunk 1 above)
Comment (dorusomcutean) [reply routes/orders/order.model.js:120]: 2h wasted of my life to find this... https://stackoverflow.com/questions/35626040/findoneandupdate-used-with-returnnewdocumenttrue-returns-the-original-document last answer... 
Basically, it needs that to return the updated document, not the original one...

**Comment by last1here** [comment]
LGTM 🚀

**Pull request merged**